registry/storage/driver/s3-aws/s3.go

```diff
@@ -1251,43 +1261,90 @@ func (d *driver) getStorageClass() *string {
 	return aws.String(d.StorageClass)
 }
 
+type completedParts []*s3.CompletedPart
+
+func (a completedParts) Len() int           { return len(a) }
+func (a completedParts) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a completedParts) Less(i, j int) bool { return *a[i].PartNumber < *a[j].PartNumber }
```

Comment:
I like keeping type implementations grouped together instead of interleaving them - happy to change

Reply:
I'm not strongly attached to either, but as this was a straight move, without changes, it could make sense to do that move in a separate commit (to make it clearer what parts of this patch are related to the optimisation, and what parts are to "assist" those changes).

Reply:
This is a fair point @thaJeztah I will put it back so it doesn't pop up in the diff, but as I said my personal preference is to have type implementation grouped together rather than interleaving them with other types 😄